apps/condo/domains/common/components/MultipleFileUpload.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { useState, useEffect, useReducer, useMemo, useRef } from 'react'
+import { Upload } from 'antd'
+import { Button } from '@condo/domains/common/components/Button'
+import { useIntl } from '@core/next/intl'
+import { UploadOutlined, DeleteFilled } from '@ant-design/icons'
+
+import { MAX_UPLOAD_FILE_SIZE } from '@condo/domains/common/constants/uploads'
+
+import { UploadRequestOption } from 'rc-upload/lib/interface'
+import { File } from '../../../schema'
+import { UploadFile } from 'antd/lib/upload/interface'
+import { isEmpty } from 'lodash'
+
+type DBFile = {
+    id: string
+    file: File
+}
+type UploadListFile = UploadFile & {
+    id: string
+}
+
+type Module = {
+    useCreate: (attrs, onComplete) => (attrs) => Promise<DBFile>
+    useUpdate: (attrs, onComplete) => (update, attrs) => Promise<DBFile>
+    useSoftDelete: (attrs, onComplete) => (state, attrs) => Promise<unknown>
+}
+
+const reducer = (state, action) => {
+    const { type, payload: file } = action
+    switch (type) {
+        case 'delete':
+            return {
+                ...state,                    
+                added: [...state.added].filter(addFile => addFile.id !== file.id),
+                deleted: [...state.deleted, file],
+            }
+        case 'add':
+            return {
+                ...state,
+                added: [...state.added, file],
+            }
+        default:
+            throw new Error(`unknown action ${type}`)
+    }
+}
+
+const convertFilesToUploadFormat = (files: DBFile[]): UploadListFile[] => {
+    if (isEmpty(files)) {
+        return []
+    }
+    const uploadFiles = files.map(({ id, file }) => {
+        const fileInList = {
+            uid: file.id,
+            id,
+            name: file.originalFilename,
+            status: null,
+            url: file.publicUrl,
+        }
+        fileInList.status = 'done'
```

Comment:
Why you don't set it in a line above?

Reply:
Because of type script bug. TS refuses to assign string to enum defined type  Has the same issue in chessboard.

Reply:
Will add Todo and cast "as"